gym_minigrid/wrappers.py

```diff
@@ -241,12 +241,78 @@ def observation(self, obs):
             COLOR_TO_IDX['red'],
             env.agent_dir
         ])
-
+        agent_dir_one_hot = np.zeros(4)
+        agent_dir_one_hot[env.agent_dir] = 1.
         return {
             'mission': obs['mission'],
-            'image': full_grid
+            'image': full_grid,
+            'robot_obs': np.array([*agent_dir_one_hot])
+        }
+
+class EasyModeWrapper(gym.core.ObservationWrapper):
+    """
+    Fully observable gridworld using a compact grid encoding
+    """
+
+    def __init__(self, env):
+        super().__init__(env)
+
+        self.observation_space.spaces["image"] = spaces.Box(
+            low=0,
+            high=255,
+            shape=(self.env.width, self.env.height, 3),  # number of cells
+            dtype='uint8'
+        )
+
+    def observation(self, obs):
+        env = self.unwrapped
+        full_grid = env.grid.encode()
+        full_grid[env.agent_pos[0]][env.agent_pos[1]] = np.array([
+            OBJECT_TO_IDX['agent'],
+            COLOR_TO_IDX['red'],
+            env.agent_dir
+        ])
+        agent_dir_one_hot = np.zeros(4)
+        agent_dir_one_hot[env.agent_dir] = 1.
+        return {
+            'mission': "go to goal", #obs['mission'],
+            'visual_obs': obs['goal'],
+            'robot_obs': np.array([*env.agent_pos, *agent_dir_one_hot])
         }
 
+class NoLanguageWrapper(gym.core.ObservationWrapper):
+    """
+    Fully observable gridworld using a compact grid encoding
+    """
+
+    def __init__(self, env):
+        super().__init__(env)
+
+        self.observation_space.spaces["image"] = spaces.Box(
+            low=0,
+            high=255,
+            shape=(self.env.width, self.env.height, 3),  # number of cells
+            dtype='uint8'
+        )
+
+    def observation(self, obs):
+        env = self.unwrapped
+        full_grid = env.grid.encode()
+        full_grid[env.agent_pos[0]][env.agent_pos[1]] = np.array([
+            OBJECT_TO_IDX['agent'],
+            COLOR_TO_IDX['red'],
+            env.agent_dir
+        ])
+        agent_dir_one_hot = np.zeros(4)
+        agent_dir_one_hot[env.agent_dir] = 1.
+        return {
+            'mission': "go to goal", #obs['mission'],
+            'image': full_grid,
+            # 'robot_obs': obs['goal']
+            'robot_obs': np.array([*obs['goal'], *agent_dir_one_hot])
+        }
```

Comment:
As above I would suggest subclassing. To access the full grid+agent direction you can call the parent method.

```python
class NoLanguageWrapper(FullyObsWrapper):

    def observation(self, obs):
        obs_dict = super().observation(obs)
        ... # modify obs_dict as necessary to add the target position
        return obs_dict
```

Also updating the docstring would be helpful.